Core/Src/main.c

```diff
@@ -55,7 +57,12 @@ void SystemClock_Config(void);
 
 /* Private user code ---------------------------------------------------------*/
 /* USER CODE BEGIN 0 */
-
+//Single potentiometer on CH0
+uint8_t txData[2];
```

Comment:
Why not initialize the array here, during the declaration, so then you can make `txData` const instead of setting it in main?